python/lsst/pipe/tasks/transformMeasurement.py

```diff
@@ -152,38 +138,85 @@ class RunTransformTask(pipeBase.CmdLineTask):
     - For each input dataRef, reads the SourceCatalog, WCS and calibration from the
       repository and executes TransformTask.
 
-    This can be sub-tasked to support whatever dataset types or sources for
-    the WCS and calibration information are required.
+    This is not a fully-fledged command line task: it requires specialization to a particular
+    source type.  That can be conveniently provided by makeTransformCmdLineTask.
     """
-    ConfigClass = RunTransformConfig
     RunnerClass = pipeBase.ButlerInitializedTaskRunner
-    _DefaultName = "transformMeasurement"
 
     def __init__(self, *args, **kwargs):
         pipeBase.CmdLineTask.__init__(self, *args, config=kwargs['config'], log=kwargs['log'])
-        if self.config.measType == "SingleFrame":
-            pluginRegistry = measBase.sfm.SingleFramePlugin.registry
-        elif self.config.measType == "Forced":
+        if self.config.wasForced:
             pluginRegistry = measBase.forcedMeasurement.ForcedPlugin.registry
+        else:
+            pluginRegistry = measBase.sfm.SingleFramePlugin.registry
         self.makeSubtask('transform', pluginRegistry=pluginRegistry,
                          measConfig=kwargs['butler'].get(self.config.measConfig).measurement.value,
-                         inputSchema=kwargs['butler'].get("src_schema").schema)
+                         inputSchema=kwargs['butler'].get(self.config.sourceType + "_schema").schema)
 
```

Comment:
We can't assume that the measurement configuration will always be in `.measurement`; we'll also need to handle e.g. the two measurements run in the `.calibrate` subtask of `ProcessCcdTask` (i.e. `root.calibrate.initialMeasurement` and `root.calibrate.measurement`).  The `initialMeasurement` run is a little weird, though, as it's added into the calibrate catalog dataset ("icSrc") via a SchemaMapper with a prefix, so it may not work with your lower-level code at all.  I'd be content with ignoring it, but we do need to be able to handle `root.calibrate.measurement`.

Given that `ProcessCcdTask` generates two different catalogs, I'm torn as to whether we should have a single transform CmdLineTask to handle both of them or a separate one for each catalog.  I suspect the latter is probably easier.

By the way, `ProcessCoaddTask` may well have a `CalibrateTask` stage as well, but if so, that's a historical artifact that will go away with the HSC merges, and it's not worth worrying about here.